front/lib/resources/storage/models/groups.ts

```diff
@@ -0,0 +1,62 @@
+import type {
+  CreationOptional,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+} from "sequelize";
+import { DataTypes, Model } from "sequelize";
+
+import { Workspace } from "@app/lib/models/workspace";
+import { frontSequelize } from "@app/lib/resources/storage";
+
+export class GroupModel extends Model<
+  InferAttributes<GroupModel>,
+  InferCreationAttributes<GroupModel>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare name: string;
+  declare isWorkspace: boolean;
+
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+}
+GroupModel.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    isWorkspace: {
```

Comment:
I wrote the SCHEMA. And will self criticize it that we should use a status or type vs a boolean.

We will likely have other group types. Eg if we want a system group that holds the system vault for connections + give the system API keys system group rights.

So Maybe `type = "regular" | "workspace" | "system"` is desirable.

Noob SQL question can we enforce unique index for type="workspace" per workspace?

Reply:
Trying with a beforeSave hook 👍🏻

Reply:
WDYT as is?